content/posts/2020/12/libplanet-0.10/index.eng.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Libplanet 0.10 Released
+date: 2020-12-08
+authors: [suho.lee]
+translators: [kidon.seo]
+---
+
+Hello Everyone! We have released the tenth minor version of [Libplanet], [Version 0.10][0.10.0].
+
+Libplanet is a common library that solves game implementation problems such as P2P communication and data synchronization when creating online multiplayer games that run on distributed P2P.
+
+In this version, Libplanet has made many changes both inside and outside the interface, including the use of [<abbr title="Merkle–Patricia Trie">MPT</abbr>][MPT] to manage Libplanet's state and additional APIs for asset management.
+
+This piece addresses key major changes in Version 0.10.
+
+[Libplanet]: https://libplanet.io/
+[0.10.0]: https://github.com/planetarium/libplanet/releases/tag/0.10.0
+[MPT]: https://eth.wiki/en/fundamentals/patricia-tree 
+
+Result State is Now Included in [`Block<T>.Hash`]
+--------------------------------------------------
+
+Until now, [`Block<T>`] did not have any information about the state of the block. And while you could derive the state from the block itself, the only way to verify the consistency of the block's state was to run the action directly. However, `Block<T>.Hash` is now derived not only with information about the block, but also with [`Block<T>.StateRootHash`], which is the hash value of the evaluated actions in the block. As before, the hash value without the evaluated actions is placed in the [`Block<T>.PreEvaluationHash`] property.
+
+[`Block<T>`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html
+[`Block<T>.PreEvaluationHash`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_PreEvaluationHash
+[`Block<T>.Hash`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_Hash
+[`Block<T>.StateRootHash`]: https://docs.libplanet.io/0.10.2/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_StateRootHash
+
+[`Block<T>.TotalDifficulty`]
+------------------------------
+
+Originally, the block height ([`Block<T>.Index`]) was the only criteria for selecting a canonical chain in the blockchain. However, since it is common for a network to have multiple blocks of the same height at a similar rate, agreements were often only made locally. This was due to the availability of multiple options to choose from when there are blocks with the same height. To fix this issue, the new version includes [`Block<T>.TotalDifficulty`] in the selection criteria to always have one option for the canonical chain. `Block<T>.TotalDifficulty` property is the difficulty level of all blocks from itself to the genesis block. This has also improved security to some level because you can't force your chain as canonical simply by quickly stacking low-difficulty blocks.
```

Comment:
```suggestion
Originally, the block height ([`Block<T>.Index`]) was the only criteria for selecting a canonical chain in the blockchain. However, since it is common for a network to have multiple blocks of the same height at a similar rate, agreements were often only made locally, causing the canonical chain to be indeterministic. To fix this ambiguity, the new version includes [`Block<T>.TotalDifficulty`] in the selection criteria to always have one option for the canonical chain. `Block<T>.TotalDifficulty` property is the difficulty level of all blocks from itself to the genesis block. This has also improved security to some level because you can't force your chain as canonical simply by quickly stacking low-difficulty blocks.
```